### PR TITLE
[chore] Update tests for confmap strict type change

### PR DESCRIPTION
**Pull request by TylerHelmuth**
Make room for https://github.com/open-telemetry/opentelemetry-collector/pull/10554

### CODE REVIEW

**Review by codeboten — APPROVED** [review]

File: processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ receivers:
 processors:
 
   probabilistic_sampler/traces:
-    sampling_percentage: +Inf
+    sampling_percentage: 999_999_999_999_999_999_999_999_999_999_999_999_999
```
Comment (codeboten) [inline processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml:7]: is `+Inf` no longer supported?

File: connector/failoverconnector/testdata/config.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 failover:
 
 failover/default:
-  priority_levels: [traces]
+  priority_levels:
+    - [traces]
```
Comment (codeboten) [inline connector/failoverconnector/testdata/config.yaml:5]: should this be `- traces`?

File: processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml (hunk 1 above)
Comment (TylerHelmuth) [reply processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml:7]: I wasn't able to find a way to get it interpreted as a float. I tried `!!float +Inf` and got this error message:

```
cannot resolve the configuration: cannot retrieve the configuration: yaml: cannot decode !!str `+Inf` as a !!float"
```

File: connector/failoverconnector/testdata/config.yaml (hunk 2 above)
Comment (TylerHelmuth) [reply connector/failoverconnector/testdata/config.yaml:5]: The field is a [[][]](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1b4f0451d1df1c2d66cfdb4daa3345fa29cae133/connector/failoverconnector/config.go#L22)

**Review by songy23 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

File: processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml (hunk 1 above)
Comment (mx-psi) [reply processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml:7]: From https://yaml.org/spec/1.2.2/#1032-tag-resolution, `+.Inf` should work

Comment (mx-psi) [reply processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml]: Updated this on dde0255f442ae162aa08ef58598f035f79089807

Comment (codeboten) [reply processor/probabilisticsamplerprocessor/testdata/invalid_inf.yaml]: Thanks!

**Pull request merged**